mon-pix/tests/integration/components/qrocm-dep-solution-panel-test.js

```diff
@@ -0,0 +1,243 @@
+          this.set('answer', answer);
+
+          // when
+          await render(hbs`{{qrocm-dep-solution-panel challenge=challenge answer=answer solution=solution}}`);
+        });
+
+        it('should display the correct answer in green bold', function() {
+          // then
+          const firstAnswerInput = findAll(ANSWER)[0];
+          expect(firstAnswerInput.classList.contains('correction-qroc-box-answer--correct')).to.be.true;
+        });
+
+        it('should not have solution block', function() {
+          // then
+          const solutionBlockList = findAll(SOLUTION_BLOCK);
+          expect(solutionBlockList).to.have.lengthOf(0);
+        });
+      });
+
+      describe('When there is no answer', function() {
+        const EMPTY_DEFAULT_MESSAGE = 'Pas de réponse';
+
+        beforeEach(async function() {
+          // given
+          const answer = EmberObject.create({
+            id: 'answer_id',
+            value: 'key1: \'\' key2: \'\'',
+            result: classByResultKey.aband,
+            assessment,
+            challenge
+          });
+          this.set('answer', answer);
+
+          // when
+          await render(hbs`{{qrocm-dep-solution-panel challenge=challenge answer=answer solution=solution}}`);
+        });
+
+        it('should display one solution in bold green', async function() {
+          // then
+          const noAnswerSolutionBlockList = findAll(SOLUTION_BLOCK);
+          expect(noAnswerSolutionBlockList).to.have.lengthOf(1);
+        });
+
+        it('should display the empty answer with the default message "Pas de réponse" in italic', async function() {
+          // then
+          const firstAnswerInput = findAll(ANSWER)[0];
+          expect(firstAnswerInput).to.exist;
+          expect(firstAnswerInput.value).to.equal(EMPTY_DEFAULT_MESSAGE);
+          expect(firstAnswerInput.classList.contains('correction-qroc-box-answer--aband')).to.be.true;
+        });
+      });
+
+      describe('When the answer is wrong', function() {
+        beforeEach(async function() {
+          // given
+          const answer = EmberObject.create({
+            id: 'answer_id',
+            value: 'key1: \'wrongAnswer1\' key2: \'wrongAnswer2\'',
+            result: classByResultKey.ko,
+            assessment,
+            challenge
+          });
+          this.set('answer', answer);
+
+          // when
+          await render(hbs`{{qrocm-dep-solution-panel challenge=challenge answer=answer solution=solution}}`);
+        });
+
+        it('should display one solution in bold green', async function() {
+          // then
+          const wrongSolutionBlock = find(SOLUTION_BLOCK);
+          const wrongSolutionText = find(SOLUTION_TEXT);
+
+          expect(wrongSolutionBlock).to.exist;
+          expect(wrongSolutionText).to.exist;
+        });
+
+        it('should display the wrong answer in standard style since we do not handle single wrong answer yet', async function() {
+          // then
```

Comment:
Le `since we do not handle single wrong answer yet` me parait difficile à maintenir. Vous penserez à l'enlever le jour où ce sera le cas? 😏

Reply:
Si on fonctionne bien en TDD, normalement oui :)